src/PowerShellEditorServices/Language/LanguageService.cs

```diff
@@ -646,34 +653,43 @@ public FindOccurrencesResult FindSymbolsInFile(ScriptFile scriptFile)
                 },
                 true);
 
-            if (foundAsts != null && foundAsts.Any())
+            if (foundAsts == null || !foundAsts.Any())
             {
-                // of all the function definitions found, return the innermost function
-                // definition that contains `lineNumber`
-                funcDefnAst = foundAsts.Cast<FunctionDefinitionAst>().Aggregate((x, y) =>
-                {
-                    if (x.Extent.StartOffset >= y.Extent.StartOffset && x.Extent.EndOffset <= x.Extent.EndOffset)
-                    {
-                        return x;
-                    }
-
-                    return y;
-                });
+                helpLocation = null;
+                return null;
+            }
 
-                // TODO use tokens to check for non empty character instead of just checking for line offset
-                if (funcDefnAst.Body.Extent.StartLineNumber == lineNumber - 1)
+            // of all the function definitions found, return the innermost function
+            // definition that contains `lineNumber`
+            foreach (FunctionDefinitionAst foundAst in foundAsts.Cast<FunctionDefinitionAst>())
```

Comment:
Do you need to use Cast? Could you just use something like `as`?

Reply:
`as` won't work since it's an `IEnumerable<Ast>`. And even with covariant collections, every `IEnumerable<Ast>` is not an `IEnumerable<FunctionDefintionAst>`. As far as I know that means we need to use LINQ's `Cast<T>` or `OrType<T>`. Since we are expecting all the `Ast`s returned, I chose `Cast<T>` in this instance. It all stems from the fact that `Ast.FindAll()` can't return a `IEnumerable<T> where T : Ast`, just `IEnumerable<Ast>`, which I suppose is fair enough -- the kind of limitation I constantly see in C#'s type system (trying to be clever about these types using generic constraints has only ever bitten me later on, sadly)

Reply:
The `as` happens on line 642.